File: sol/solution/0001-0100/0007. Reverse Integer/Solution.erl
-spec reverse(X :: integer()) -> integer().
reverse(X) ->
    reverse(X, 0).
reverse(0, Acc) ->
    if Acc < -2147483648; Acc > 2147483647 -> 0;
       true -> Acc
    end;
reverse(X, Acc) ->
    Pop = X rem 10,
    NewX = X div 10,
    NewAcc = Acc * 10 + Pop,
    if NewAcc < -2147483648; NewAcc > 2147483647 -> 0;
       true -> reverse(NewX, NewAcc)
    end.

File: sol/solution/0001-0100/0007. Reverse Integer/Solution.ex
defmodule Solution do
  @spec reverse(x :: integer) :: integer
  def reverse(x) do
    x
    |> Integer.digits() 
    |> Enum.reverse() 
    |> Integer.undigits()
    |> verify_bounds()
  end
  defp verify_bounds(x) when x > 2147483647, do: 0
  defp verify_bounds(x) when x < -2147483648, do: 0
  defp verify_bounds(x), do: x
end